tcms/rpc/tests/test_testcase.py

```diff
@@ -522,16 +621,47 @@ def test_add_component_from_another_product_is_not_allowed(self):
             self.rpc_client.TestCase.add_component(self.test_case.pk, self.bad_component.name)
 
 
-class TestCaseAddComment(APITestCase):
+class TestRemoveComponentPermission(APIPermissionsTestCase):
+    permission_label = 'testcases.delete_testcasecomponent'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+        self.test_case = TestCaseFactory()
+        self.good_component = ComponentFactory(product=self.test_case.category.product)
+        self.bad_component = ComponentFactory(product=self.test_case.category.product)
+        self.test_case.add_component(self.good_component)
+        self.test_case.add_component(self.bad_component)
+
+    def verify_api_with_permission(self):
+        self.rpc_client.TestCase.remove_component(self.test_case.pk,
+                                                  self.bad_component.pk)
+        result = TestCaseComponent.objects.filter(case=self.test_case)
```

Comment:
`self.test_case.component` is a Django queryset (b/c there is a many-to-many relationship between these fields), which means it has methods .all(), .count() and .first().  So you can fetch the results after the deletion directly from the DB instead of going through the API layer.